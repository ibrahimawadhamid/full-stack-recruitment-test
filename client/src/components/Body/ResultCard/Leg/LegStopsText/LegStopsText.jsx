import React, {Component} from 'react';
import BpkText from "bpk-component-text";
import STYLES from './LegStopsText.scss';

const classes = className => STYLES[className] || 'UNKNOWN';

class LegStopsText extends Component {
  render() {
    let directFlight = <BpkText textStyle="base"
                                className={classes('rightAlign') + ' ' + classes('bottomAlign') + ' ' + classes('greenFill')}>Direct</BpkText>;
    if (this.props.legInfo["stopsStations"].length) {
      directFlight =
        <BpkText id={this.props.itineraryId + this.props.legInfo["Id"] + "moreStops"}
                 onClick={this.props.openAdditionalStopsPopover} textStyle="base"
                 className={classes('rightAlign') + ' ' + classes('bottomAlign') + ' ' + classes('redFill')}>
          {this.props.legInfo["stopsStations"].length + (this.props.legInfo["stopsStations"].length == 1 ? " stop " : " stops ")}
        </BpkText>;
    }
    return (
      <div>
        {directFlight}
      </div>
    );
  };
}

export default LegStopsText;
