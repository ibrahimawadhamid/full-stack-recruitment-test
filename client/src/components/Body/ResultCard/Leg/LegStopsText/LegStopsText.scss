@import '~bpk-mixins';

.relativePosition {
  position: relative;
}

.rightAlign {
  float: right;
}

.leftAlign {
  float: left;
}

.bottomAlign {
  float: bottom;
}

.bottomMargin {
  margin-bottom: 1.5rem;
}

.grayFill500 {
  color: $bpk-color-gray-500;
}
.grayFill300 {
  color: $bpk-color-gray-300;
}

.greenFill {
  color: $bpk-color-green-500;
}
.redFill {
  color: $bpk-color-red-500;
  text-decoration: underline;
}

.arrowGrayColor {
  fill: $bpk-color-gray-300;
}
