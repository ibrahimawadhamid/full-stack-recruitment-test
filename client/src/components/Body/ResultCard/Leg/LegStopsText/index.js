import LegStopsText from './LegStopsText';

export default LegStopsText;
