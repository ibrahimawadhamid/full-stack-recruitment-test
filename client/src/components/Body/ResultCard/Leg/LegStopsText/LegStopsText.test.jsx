import React from 'react';
import TestRenderer from 'react-test-renderer';

import LegStopsText from './LegStopsText';

describe('LegStopsText', () => {
    it('should render correctly', () => {
        const tree = TestRenderer.create(<LegStopsText legInfo={{ stopsStations: ['a', 'b'] }} />).toJSON();

        expect(tree).toMatchSnapshot();
    });
});