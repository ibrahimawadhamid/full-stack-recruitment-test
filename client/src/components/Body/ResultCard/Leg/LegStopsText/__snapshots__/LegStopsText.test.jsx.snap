// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LegStopsText should render correctly 1`] = `
<div>
  <span
    className="bpk-text bpk-text--base rightAlign bottomAlign redFill"
    id="NaNmoreStops"
  >
    2 stops 
  </span>
</div>
`;
